Add AGENTS.md file to robottelo

ColeHiggins2 opened the pull request:
As part of the effort to add AGENTS.md files to our core 3 repositories(SAT-39763), This add AGENTS.md to robottelp.

This was generated by using cursor and the Claude sonnet 4.5 model. The goal is to shape AI interaction with this repo, so that you get a more useful and uniform output. Hopefully improving the quality of our AI written PRs. 

Please add any comments and suggestions for updates. 

I would like to get as many team members involved as possible :)

LadislavVasina1 commented on AGENTS.md:
```diff
@@ -0,0 +1,952 @@
+ak = target_sat.api.ActivationKey(organization=org).create()
+
+# Using UI session
+with target_sat.ui_session() as session:
```
I find it a good practice to do Org/Loc selection as first step in the UI tests.
```suggestion
with target_sat.ui_session() as session:
	session.organization.select('ORG_NAME')
	session.location.select('LOC_NAME')
```

LadislavVasina1 commented on AGENTS.md:
````diff
@@ -0,0 +1,952 @@
+### Pattern 3: CLI Test
+
+```python
+@pytest.mark.tier1
````
```suggestion
```

LadislavVasina1 commented on AGENTS.md:
````diff
@@ -0,0 +1,952 @@
+
+```python
+@pytest.mark.e2e
+@pytest.mark.tier3
````
```suggestion
```

LadislavVasina1 commented on AGENTS.md:
````diff
@@ -0,0 +1,952 @@
+@pytest.mark.rhel_ver_list([9, 10])
+def test_with_rhel9_and_10(rhel_contenthost):
+    ...
+```
````
```suggestion
# If there is no need for multiple RHEL versions to be tested, but RHEL host is still needed for the sake of the test.

from robottelo.config import settings
@pytest.mark.rhel_ver_match([settings.content_host.default_rhel_version])
def test_with_default_rhel(rhel_contenthost):
	...
```

LadislavVasina1 commented on AGENTS.md:
````diff
@@ -0,0 +1,952 @@
+@pytest.mark.tier1  # High
+@pytest.mark.tier2  # Medium
+@pytest.mark.tier3  # Low
+```
````
```suggestion
```

LadislavVasina1 commented on AGENTS.md:
```diff
@@ -0,0 +1,952 @@
+@pytest.mark.rhel_ver_match(r'^[\d]+$')
+
+# Match RHEL 9 including FIPS
+@pytest.mark.rhel_ver_match(r'^9')  # Matches 9, 9_fips
```
```suggestion
@pytest.mark.rhel_ver_match(r'^9')  # Matches 9, 9_fips

# Match default RHEL version which is in the settings
@pytest.mark.rhel_ver_match([settings.content_host.default_rhel_version])
```

ogajduse replied:
Let's discuss this in #20565

LadislavVasina1 commented on AGENTS.md:
```diff
@@ -0,0 +1,952 @@
+- `:id:` - Unique test UUID
+- `:steps:` - Test execution steps
+- `:expectedresults:` - Expected outcome
+
```
```suggestion
** Optional fields**
- `:Verifies:` - When the test verifies a Bug (Use as :Verifies: SAT-12345) (Formely there was :BZ: tag, don't use that anymore) 
```

ogajduse commented on AGENTS.md:
```diff
@@ -0,0 +1,952 @@
+- `test_negative_create_ak_with_invalid_name()`
+- `test_upgrade_content_view_promotion()`
+
+### Docstring Style
```
Here I'd point the agent to `testimony.yaml` so it knows what the choices for each [token](https://github.com/SatelliteQE/testimony/blob/master/docs/index.rst#test-case-docstring-format) are. `testimony.yaml` can also be the source of truth for what tokens are required.

ogajduse commented on AGENTS.md:
```diff
@@ -0,0 +1,952 @@
+- Use `module` scope for expensive fixtures (Satellite, manifests)
+- Use `function` scope for test-specific fixtures
+- Clean up resources in fixture teardown (use `yield`)
+- Parametrize fixtures using `@pytest.fixture(params=[...])` when prompted
```
Pardon my ignorance, what does `when prompted` mean in this context?

ogajduse commented on AGENTS.md:
```diff
@@ -0,0 +1,952 @@
+    'id': task_id,
+    'timeout': DEFAULT_TIMEOUT
+})
+assert result['result'] == 'success'
```
This is quite unusual, and we use it only in one test. Keep it or remove it?

ogajduse commented on AGENTS.md:
```diff
@@ -0,0 +1,952 @@
+
+**Solution**:
+- Check fixture is defined in `conftest.py` or fixture file
+- Ensure fixture scope matches test scope
```
Do we have test scopes? 🤔

ogajduse commented on AGENTS.md:
````diff
@@ -0,0 +1,952 @@
+)
+```
+
+#### 5. **UI Test Element Not Found**
````
@LadislavVasina1, Mind taking a look at all UI-related items in this doc?

ogajduse commented on AGENTS.md:
```diff
@@ -0,0 +1,952 @@
+
+*   **Test Configuration:** The application uses multiple YAML files for configuration:
+    - `robottelo.yaml`: Main test configuration
+    - `broker_settings.yaml`: Broker/VM configuration
```
`broker_settings.yaml` is an optional config that does not need to be present; broker can use `~/.broker/broker_settings.yaml`.

ColeHiggins2 commented on AGENTS.md:
```diff
@@ -0,0 +1,952 @@
+def test_positive_create_activation_key(module_org, module_target_sat):
+    """Create activation key with valid name
+    
+    :id: 1a2b3c4d-5e6f-7a8b-9c0d-1e2f3a4b5c6d
```
ill add as a comment

ogajduse replied:
To be resolved in #20557

ColeHiggins2 commented on AGENTS.md:
```diff
@@ -0,0 +1,952 @@
+**DO ✅**:
+- Use `module` scope for expensive fixtures (Satellite, manifests)
+- Use `function` scope for test-specific fixtures
+- Clean up resources in fixture teardown (use `yield`)
```
i was curious about this myself... do we actually do this?

ColeHiggins2 commented on AGENTS.md:
```diff
@@ -0,0 +1,952 @@
+**DON'T ❌**:
+- Don't create new Satellite instances per test (use `module_target_sat`)
+- Don't hard-code values in fixtures (use `gen_string()`)
+- Don't skip fixture cleanup (always delete created resources)
```
ill remove the last part. should we remove the whole thing?

ogajduse replied:
I'd be in favor of keeping this cleanup-related instruction there, if no one else objects. It might look like I have a split personality here, but this is something we should have a discussion about in the broader team. 

But for starters, I'd keep it there, and let's see what people say; maybe we'll see a PR removing this instruction from the list in the future. That will allow more focused discussion on this topic than here.

Feel free to resolve this thread. :slightly_smiling_face:

ogajduse replied:
To be resolved in [#20561](https://github.com/SatelliteQE/robottelo/issues/20561)

ColeHiggins2 commented on AGENTS.md:
````diff
@@ -0,0 +1,952 @@
+```python
+@pytest.mark.e2e              # End-to-end tests
+@pytest.mark.stubbed          # Not yet implemented
+@pytest.mark.destructive      # Modifies Satellite config and needs satellite teardown
````
what do you think we should add?

ColeHiggins2 commented:
@JacobCallahan @ogajduse @LadislavVasina1 Please let me know what you think! I addressed most of the comments and added comments on ones I was unsure of

vsedmik commented on AGENTS.md:
```diff
@@ -0,0 +1,924 @@
+- `function`: Per test function (default)
+- `module`: Per test module
+- `session`: Per test session
+- `class`: Per test session
```
```suggestion
- `class`: Per test class
```
Also, `class` should fit between `function` and `module` if we want to order the particular types per their scope.

vsedmik commented on AGENTS.md:
```diff
@@ -0,0 +1,952 @@
+- `.ready()`: Returns setup data after upgrade
+- Markers: `@pytest.mark.{feature}_upgrades`
+
+### Old Upgrade Pattern (Legacy)
```
I would omit the legacy scenarios.

vsedmik commented on AGENTS.md:
```diff
@@ -0,0 +1,924 @@
+
+#### 2. **Test Hangs During Execution**
+
+**Problem**: Test hangs indefinitely
```
```suggestion
**Problem**: Test exits with timeout for an action
```

vsedmik commented on AGENTS.md:
```diff
@@ -0,0 +1,924 @@
+    wait_for(lambda: session.activationkey.is_displayed, timeout=30)
+    
+    # Ensure page is stable
+    session.browser.plugin.ensure_page_safe(timeout='10s')
```
I think this should be fixed on the AirGun level, not in test.

ogajduse commented on AGENTS.md:
```diff
@@ -0,0 +1,952 @@
+
+Robottelo supports two upgrade testing patterns:
+
+### New Upgrade Pattern (Recommended)
```
gentle ping on this

ogajduse commented on AGENTS.md:
```diff
@@ -0,0 +1,952 @@
+*   **Pre-commit Hooks:** The project can use `pre-commit` to run checks before committing.
+    - Run manually: `pre-commit run --all-files`
+
+### Configuration
```
Here is a PR with suggested content. Since (mis)configuration is often a source of problems, I thought we should document it well.

https://github.com/ColeHiggins2/robottelo/pull/219

If you merge the PR as-is, the changes will propagate here in this PR.

ogajduse reviewed (changes requested):
I'd like to ask you to go through my comments and provide a feedback on them. Given how large the patch is, I'd like to have all my comments resolved before we merge it. Feel free to resolve the straightforward ones where no further discussion is needed. And the same for comments that were already addressed by making the requested change. This will make it easier to re-review the PR.

ogajduse commented on AGENTS.md:
```diff
@@ -0,0 +1,952 @@
+- **Location**: `robottelo/`
+- **Components**:
+  - **API helpers**: `robottelo/api/` (Nailgun entities)
+  - **CLI helpers**: `robottelo/cli/` (Hammer command wrappers)
```
Future TODO: #20556

ogajduse commented on AGENTS.md:
````diff
@@ -0,0 +1,952 @@
+```
+
+**Required Fields**:
+- `:id:` - Unique test UUID
````
To be done in #20557

ogajduse commented on AGENTS.md:
````diff
@@ -0,0 +1,952 @@
+```
+
+**Required Fields**:
+- `:id:` - Unique test UUID
````
This could be looked at as part of #20557

ogajduse commented on AGENTS.md:
````diff
@@ -0,0 +1,952 @@
+    ).create()
+```
+
+**Organization Fixtures** (`pytest_fixtures/component/taxonomy.py`):
````
@ColeHiggins2 I'd like to get it factually correct. Can you take a look at this one before merging?

ogajduse commented on AGENTS.md:
to be resolved in #20560

ogajduse commented on AGENTS.md:
```diff
@@ -0,0 +1,924 @@
+Examples:
+- `test_positive_create_activation_key_with_cv()`
+- `test_negative_create_ak_with_invalid_name()`
+- `test_upgrade_content_view_promotion()`
```
To be done in #20562

ogajduse commented on AGENTS.md:
```diff
@@ -0,0 +1,952 @@
+- `target_sat`: A Satellite instance for the test
+- `module_target_sat`: Module-scoped Satellite instance
+- `function_sca_manifest`: Function-scoped SCA manifest
+- `rhel_contenthost`: RHEL content host for testing
```
To be looked at as part of #20565

ogajduse reviewed (commented):
I managed to close almost all threads. I left those that I found important open.

ogajduse commented:
I created https://github.com/SatelliteQE/robottelo/issues/20567 to track TODOs for AGENTS.md.
